konduit-serving-io/konduit-serving-camera/src/main/java/ai/konduit/serving/camera/step/capture/CameraFrameCaptureStep.java

```diff
@@ -19,22 +19,38 @@
 
 import ai.konduit.serving.annotation.json.JsonName;
 import ai.konduit.serving.pipeline.api.step.PipelineStep;
+import io.swagger.v3.oas.annotations.media.Schema;
 import lombok.Builder;
 import lombok.Data;
 import org.nd4j.shade.jackson.annotation.JsonProperty;
 
+// TODO: not sure if it's relevant here but maybe we can later add a variable to specify the frames per second
+//  or a way to specify max number of inferences per second. This can potentially save us some compute.- SHAMS
```

Comment:
Currently it's basically polling based - "grab a frame when we execute the pipeline".
I will be writing an ADR shortly for "loop internally + return last output on query" style pipeline operations instead.
That will have the delay / frames-per-second configuration, and the number of pipeline executions per second will no longer depend on the number of clients querying the predict endpoint.

Reply:
Ah okay so it's already planned. I'll leave the todo here then.